storage/rocksdb/rdb_compact_filter.h

```diff
@@ -65,6 +65,17 @@ class Rdb_compact_filter : public rocksdb::CompactionFilter {
 
       if (!m_should_delete) {
         get_ttl_duration(gl_index_id, &m_ttl_duration);
+
+        /*
+          For efficiency reasons, we lazily call GetIntProperty to get the
+          oldest snapshot time, each time we find a new index to be processed.
+        */
+        rocksdb::DB *const rdb = rdb_get_rocksdb_db();
```

Comment:
Yes, keep the same m_snapshot_timestamp for the whole compaction run. No need to update it because a slightly later time will unlikely change the outcome, so it'd be easier to keep it simple with less work done.